Signup route

RafGDev opened the pull request:
I have made the signup route more modular. I have also put mysql in it's own wrapper so we can use it over multiple files. Could someone do some testing on their machine for this specific PR.

Thanks.

RafGDev committed:
Finished signup route for backend

RafGDev committed:
Making signup route more modular

RafGDev commented on backend/lib/auth.js:
```diff
@@ -0,0 +1,91 @@
+  const salt = crypto.randomBytes(16).toString("hex");
+  const hash = crypto.createHash('sha1').update(salt + password).digest('hex');
+
+  return {hash, salt: salt};
```
Urgg, I need to change this.

RafGDev committed:
Merge branch 'dev' of https://github.com/cats-with-potatos/chat-app into loginRoute

Getting latest changes for backend

JGefroh commented:
Please integrate your changes into `server.js` instead of `index.js` so the frontend portion can still be delivered via `express`. Note that the port number has been changed to `8080`. You can start `server.js` by using `npm start`.

RafGDev committed:
Deleted index.js and put logic in server.js for npm start

RafGDev committed:
Added body-parser to parse application/x-www-form-urlencoded

jp-webdev committed:
DB Credential Adjustment, Authentication Changes

Implemented regex search patterns in the auth.js file. Added an if
statement to check whether their inputted password matches a second
inputted password -- as having a signup page with one single password
field might lead to errors.

Will have to ask the front end guys if they can get a second pass field
implemented.

Also made a minor change to db.js -- logging in with localhost will
cause Macs to attempt to connect to a non-existent socket. Using
127.0.0.1 instead.

jp-webdev committed:
Minor Naming Changes

Using camelCase for variables names -- validation_password to
validationPassword

RafGDev committed:
changed UserTable column type to varchar binary for case case sensitive usernames

RafGDev committed:
changed UserTable column type to varchar binary for case case sensitive usernames

RafGDev committed:
Merge branch 'loginRoute' of https://github.com/cats-with-potatos/chat-app into loginRoute
